adapters/smarthub/smarthub.go

```diff
@@ -60,10 +60,14 @@ func (a *adapter) getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ExtSmartHub,
 
 func (a *adapter) buildEndpointURL(params *openrtb_ext.ExtSmartHub) (string, error) {
 	endpointParams := macros.EndpointTemplateParams{
-		Host:      params.PartnerName,
 		AccountID: params.Seat,
 		SourceId:  params.Token,
 	}
+
+	if params.PartnerName != "" {
+		endpointParams.Host = params.PartnerName
+	}
+
```

Comment:
You shouldn't need to do this. You can just keep the original logic so that `Host` is always assigned the value of `params.PartnerName`. If `params.PartnerName` is an empty string, the generated URL will contain `?partnerName=&`.

Were you trying to conditionally exclude the `partnerName` query param?

I suggest either modifying an existing JSON test or adding a new JSON test where the partner name is either an empty string or not specified to verify your intended behavior.

Reply:
Yes, you're right, in our system when partner use alias-adapter we not reading parnerName param, doesn't matter whether there is data or not. I returned first variant.